docs/gettinginvolved/infrastructure/documentation.rst

```diff
@@ -0,0 +1,182 @@
+
+.. prompt:: bash $
+
+   gsutil -h x-amz-website-redirect-location:/en/${VERSION} cp - s3://clusterhq-docs/index.html </dev/null
+   gsutil -h x-amz-website-redirect-location:/en/${VERSION} cp - s3://clusterhq-docs/en/index.html </dev/null
+
+.. TODO - Specify where this is versioned. https://clusterhq.atlassian.net/browse/FLOC-1250
+
+There is an ``error.html`` uploaded to the root of the bucket. It is uploaded with:
+
+.. prompt:: bash /path/to/website/repo $
+
+   gsutil -m cp 404.html s3://clusterhq-docs/404.html
+
+
+``clusterhq-staging-docs``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This bucket is for staging changes to the main ``clusterhq-docs`` bucket.
+It is also used as part of the pre-release testing.
+
+The staging configuration is the same as the production configuration,
+with ``docs.staging.clusterhq.com`` replacing ``docs.clusterhq.com`` and ``clusterhq-staging-docs`` replacing `clusterhq-docs``.
+
+``clusterhq-dev-docs``
+~~~~~~~~~~~~~~~~~~~~~~
+
+This bucket has documentation uploaded to it from Buildbot as an intermediate stage to being copied to the above buckets.
+Buildbot will upload documentation from all builds of release branches or tags here.
+The build will be uploaded to a folded named after the python version
+(i.e. the output of ``python setup.py --version``).
+
+Configuration
+`````````````
+
+It is not configured to be publicly accessible.
+
+It has a lifecycle rule that deletes all objects older than 14 days.
+
+
+CloudFront Distributions
+------------------------
+
+We use CloudFront to provide SSL in front of the above S3 buckets.
+There are 2 CloudFront distributions, one for https://docs.clusterhq.com/ and one for https://docs.staging.clusterhq.com/.
+They are pointed at the S3 website URLs (`<bucket-name>.s3-website-us-east-1.awazonaws.com`).
+
+Configuration
+~~~~~~~~~~~~~
+The configuration of the ``docs.clusterhq.com`` distribution is:
+
+- Origin Domain Name: ``clusterhq-docs.s3-website-us-east-1.amazonaws.com``
+- Origin Path:
+- Origin ID: ``clusterhq-docs``
+- Origin Protocol Policy: HTTP Only
+- Alternate Domain Names: ``docs.clusterhq.com``
+- Viewer Protocol Policy: HTTPS Only
+- Logging: enabled
+- Bucket for Logs: ``clusterhq-logs.s3.amazonaws.com``
+- Log Prefix: ``docs.staging.clusterhq.com/cloudfront/``
+- SSL Certificate: Custom SSL Certificate: ``docs.clusterhq.com``
+- Custom SSL Client Support: Only Clients that Support Server Name Indication (SNI)
+
+The rest of the settings can be left at their defaults.
+The staging configuration is the same,
+with ``docs.staging.clusterhq.com`` replacing ``docs.clusterhq.com`` and ``clusterhq-staging-docs`` replacing `clusterhq-docs``.
+
+.. note::
+
+   We can't use an S3 origin, as redirects won't work.
+
+See the `CloudFront documetation <http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/SecureConnections.html>`_ for details on uploading SSL key material.
```

Comment:
I wonder who is responsible for generating new certificates and uploading them before the current ones expire.

Maybe there should be an issue in JIRA with a due date?  I'm not sure anyone is looking at issues in a way that makes upcoming due dates apparent, though.